src/pages/index.tsx

```diff
@@ -1,29 +1,100 @@
-import Modal from '../components/modal';
-import classNames from 'classnames';
+import Step1 from '../components/steps/step-1';
+import Step2 from '../components/steps/step-2';
+import Step3 from '../components/steps/step-3';
+import Step4Result from '../components/steps/step-4-result';
+import { isBoolean, getRandomHouseResult } from '../utils';
+import { determineIfUserWon } from '../actions';
 
 const IndexPage = (): ReactElement => {
-  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [userChoice, setUserChoice] = useState<RockPaperScissors | undefined>(undefined);
+  const [houseChoice, setHouseChoice] = useState<RockPaperScissors | undefined>(undefined);
+  const [isWin, setIsWin] = useState<boolean | undefined>(undefined);
+  const [score, setScore] = useState<number>(0);
 
-  const toggleModal = () => {
-    return setIsModalOpen(!isModalOpen);
+  // useEffect to set score from localStorage on component load
+  useEffect(() => {
+    const lsScore = localStorage.getItem('score');
+    if (lsScore) {
+      setScore(parseInt(lsScore) || 0);
+    }
+  }, []);
+
+  // useEffect to detect changes to `userChoice` and set `houseChoice` based on that
+  useEffect(() => {
+    if (userChoice) {
+      setTimeout(() => {
+        setHouseChoice(getRandomHouseResult(userChoice));
+      }, 2000);
+    }
+  }, [userChoice]);
+
+  // useEffect to detect changes to `userChoice` and set `houseChoice` based on that
+  useEffect(() => {
+    if (userChoice) {
+      setTimeout(() => {
+        setHouseChoice(getRandomHouseResult(userChoice));
+      }, 2000);
+    }
+  }, [userChoice]);
+
+  // useEffect to detect changes to `houseChoice` and set whether user has won or not
+  useEffect(() => {
+    if (houseChoice) {
+      setTimeout(() => {
+        if (userChoice && houseChoice) {
+          setIsWin(determineIfUserWon(userChoice, houseChoice));
+        }
+      }, 2000);
+    }
+  }, [houseChoice]);
+
+  // useEffect to detect changes to `isWin` and set whether user has won or not
+  useEffect(() => {
+    if (isWin) {
+      incrementScore();
+    }
+  }, [isWin]);
+
+  /**
+   * reset
+   *
+   * Reset all state and begin game again
+   */
+  const reset = () => {
+    setUserChoice(undefined);
+    setHouseChoice(undefined);
+    setIsWin(undefined);
   };
 
-  const rulesBtnClasses = classNames('absolute', 'mx-auto', 'bottom-14', 'left-1/2', 'md:right-6', 'capitalize');
-
-  return (
-    <Layout>
-      <Button className={rulesBtnClasses} onClick={toggleModal}>
-        Rules
-      </Button>
-      <Modal title="Rules" open={isModalOpen} onClose={toggleModal} center>
-        <img className="mx-auto" src="/assets/images/image-rules.svg" height={300} width={300} />
-      </Modal>
-    </Layout>
-  );
+  /**
+   * incrementScore
+   *
+   * Increment score by 1
+   */
+  const incrementScore = () => {
+    setScore(score + 1);
+    localStorage.setItem('score', JSON.stringify(score + 1));
```

Comment:
Why do you need to stringify it here?

Reply:
You normally have to do this with anything you are saving in localStorage. It is more important when you are saving a full object but TS was screaming at me for trying to save a number.